### DISCUSSION TITLE
pipeline.sendEvent()

### DISCUSSION THREAD

**Discussion by lafoletc** [gstreamer-java/gst1-java-core#282] [General]
I need to send an EOSEvent in the pipeline
As **gst_element_send_event** need an event with "transfer full", how do you instantiate and configure the event ?

do you call event.disown() ?
If yes, disown() could be added in the public constructor of each event ?

**Comment by lafoletc** [comment]
Ok, I just see the @IncRef in GstElementAPI.gst_element_send_event

**Comment by neilcsmith-net** [comment]
Yes, all `transfer full` mappings should be handled already in this way.

You can either wait for the GC to kick in, or possibly try 

```java
try (EOSEvent e = new EOSEvent(..)) {
  element.sendEvent(e);
}
```

You wouldn't want to disown these.
